coalib/parsing/FilterHelper.py

```diff
@@ -123,10 +125,20 @@ def collect_filters(args, arg_list=None, arg_parser=None):
         Instance of ArgParser that is used to parse arg list.
     :return:
         List of filters in standard filter format, i.e
-        ``[['filter_name', 'arg1', 'arg2']]``.
+        ``{'filter_name': ['arg1', 'arg2']}``.
     """
     if args is None:
         arg_parser = default_arg_parser() if arg_parser is None else arg_parser
         args = arg_parser.parse_args(arg_list)
 
-    return getattr(args, 'filter_by', None) or []
+    filters = getattr(args, 'filter_by', None) or []
+    filters = list_to_dict(filters)
+    return filters
+
+
+def list_to_dict(filters):
+    items = {}
+    for filter in filters:
+        key, value = filter[0], filter[1:]
```

Comment:
`value` should be tuple. I have already told you this. Also please rename appropriately to say `filter_name`, `args`.